app/controllers/application_controller.rb

```diff
@@ -5,6 +5,103 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
+  end
+
+
+  get '/' do
+    erb :'index'
+  end
+
+
+  get '/signup' do
+    erb :'users/create_user'
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:password] == "" || params[:email] == ""
+      redirect '/failure'
+    else
+      @user =User.create(username: params[:username], email: params[:email], password: params[:password])
+      session[:user_id] = @user.id
+      redirect '/tweets'
+    end
+  end
+
+  get '/login' do
+    erb :'users/login'
+  end
+
+  post "/login" do
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:user_id] = @user.id
+      redirect to "/tweets"
+    else
+      redirect to "/failure"
+    end
+  end
+
+    get '/tweets' do
+      erb :'tweets/tweets'
+    end
+
+    get "/logout" do
+    session.clear
+    redirect "/"
+  end
+
+  get '/tweets/new' do
+    erb :'tweets/create_tweet'
+  end
+
+  post '/tweets' do
+    if !logged_in?
+      redirect to "/failure"
+    else
+      tweet = current_user.tweets.build(params)
+      tweet.save
+    end
+    redirect to '/tweets'
+  end
+
+  get '/tweets/:id' do
+      @tweet = Tweet.find(params[:id])
+      erb :'tweets/show'
+    end
+
+  get '/tweets/:id/edit' do
+    @tweet = Tweet.find(params[:id])
+    erb :'tweets/edit'
+  end
+
+  post '/tweets/:id' do
+    tweet = Tweet.find(params[:id])
+     tweet.update(content: params[:content])
+     redirect to '/tweets'
+  end
+
+  post '/tweets/:id/delete' do
+    tweet = Tweet.find(params[:id])
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.